pkg/capabilities/capabilities_test.go

```diff
@@ -55,3 +55,55 @@ func TestGet(t *testing.T) {
 		t.Fatalf("expected Capabilities: %#v , got a different: %#v", cap, res)
 	}
 }
+func TestSetup(t *testing.T) {
+	testCases := []struct {
+		name                     string
+		allowPrivileged          bool
+		perConnectionBytesPerSec int64
+		expectedCapabilities     Capabilities
+	}{
+		{
+			name:                     "AllowPrivileged true with bandwidth limit",
+			allowPrivileged:          true,
+			perConnectionBytesPerSec: 1024,
+			expectedCapabilities: Capabilities{
+				AllowPrivileged:                        true,
+				PerConnectionBandwidthLimitBytesPerSec: 1024,
+			},
+		},
+		{
+			name:                     "AllowPrivileged false with higher bandwidth limit",
+			allowPrivileged:          false,
+			perConnectionBytesPerSec: 2048,
+			expectedCapabilities: Capabilities{
+				AllowPrivileged:                        false,
+				PerConnectionBandwidthLimitBytesPerSec: 2048,
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			defer func() {
+				capInstance.lock.Lock()
+				defer capInstance.lock.Unlock()
+				capInstance.capabilities = nil
+				capInstance.once = sync.Once{}
+			}()
```

Comment:
Would it make sense to add `capInstance.once = sync.Once{}` to the `SetForTests` and use it in this and possibly other test cases?

Reply:
Hi @bart0sh, thank you for your suggestion. I agree with your line of thinking; however, I think using SetForTests here would bypass Setup, defeating the purpose of testing how Setup actually initializes capabilities.

Am I misunderstanding what you mean?

Reply:
I thought that this code is almost the same as `SetForTests`, so you can change `SetForTests`:
```
 // SetForTests sets capabilities for tests.  Convenience method for testing.  This should only be called from tests.
-func SetForTests(c Capabilities) {
+func SetForTests(c *Capabilities) {
        capInstance.lock.Lock()
        defer capInstance.lock.Unlock()
-       capInstance.capabilities = &c
+       capInstance.capabilities = c
+       capInstance.once = sync.Once{}
 }
```

and then use it in the tests like this:
```
        for _, tc := range testCases {
                t.Run(tc.name, func(t *testing.T) {
-                       defer func() {
-                               capInstance.lock.Lock()
-                               defer capInstance.lock.Unlock()
-                               capInstance.capabilities = nil
-                               capInstance.once = sync.Once{}
-                       }()
+                       defer SetForTests(nil)
```

Changing the name of SetForTests would probably make sense too.

WDYT?

Reply:
Thank you for the clarification, @bart0sh. I see what you mean now. You're suggesting we modify the SetForTests function itself and use it to reset the capabilities for each test. I think this is a good approach.

Also, I agree that changing the name would make sense. How about `ResetCapabilitiesForTest`? This seems more descriptive of the function's purpose. I'll include the rename in the upcoming changes :+1:

Reply:
I'd suggest to name it `ResetForTest` as `capabilities.SetCapabilitiesForTest` is long and repeats `capabilities` 2 times.

Reply:
I agree, `ResetForTest` is clearer and more concise.

Reply:
Thank you for the quick update!
/lgtm

Reply:
one more thing: This code should use `ResetForTest` as well, I believe:
```
func TestGet(t *testing.T) {
	defer func() {
		capInstance.lock.Lock()
		defer capInstance.lock.Unlock()
		capInstance.capabilities = nil
		capInstance.once = sync.Once{}
	}()
```

Reply:
or should be removed as `ResetForTest` is already called in this test case.

Reply:
Oh yes I missed that test case, thank you for catching that! It has been updated in the latest changes